src/lib/agent/PolykeyClient.ts

```diff
@@ -1,8 +1,13 @@
-import { agent } from '../../../proto/js/Agent';
 import { Duplex } from 'readable-stream';
+import PeerInfo, { Address } from '../peers/PeerInfo';
+import { agent } from '../../../proto/js/Agent';
 const {
+  AddPeerRequestMessage,
```

Comment:
Just so I'm understanding this. The `Client` shouldn't need to have GRPC messages. The `Client` can send GRPC messages to the agent, but it shouldn't need to receive this. So I'm guessing then this is all about sending messages then right?

Also the HTTP API/OAuth API is again only in the agent.

Unless I'm being confused here and really these 2 files `PolykeyClient` and `PolykeyAgent` are really 2 sides of the same coin. That is the client and server sides of the agent. Because the agent is a p2p application, the agent acts as a client to other peers and as a server to other clients.

If that's the case, this needs to be made much clearer. And instead we should have client and server models that are put together into 1 `Agent` object.

The `pk` command is only a "client" in the sense of being a single-shot command. That we originally thought won't maintain session. But now it does have to due to GRPC. But we should really differentiate that. Because if that's what this file is for, it should not be under `src/lib/agent`. It needs to be somewhere else. Because it's not the _client_ side of the agent.

Reply:
so the agent _encodes_ request messages and _decodes_ response messages. Agent is opposite of this.